modules/product/src/main/java/com/opengamma/strata/product/credit/ResolvedCdsIndex.java

```diff
@@ -0,0 +1,1073 @@
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.date.DaysAdjustment;
+import com.opengamma.strata.product.ResolvedProduct;
+import com.opengamma.strata.product.common.BuySell;
+
+/**
+ * A CDS (portfolio) index, resolved for pricing.
+ * <p>
+ * This is the resolved form of {@link CdsIndex} and is an input to the pricers.
+ * Applications will typically create a {@code ResolvedCdsIndex} from a {@code CdsIndex}
+ * using {@link CdsIndex#resolve(ReferenceData)}.
+ */
+@BeanDefinition
+public final class ResolvedCdsIndex
+    implements ResolvedProduct, ImmutableBean, Serializable {
+
+  /**
+   * Whether the CDS index is buy or sell.
+   * <p>
+   * A value of 'Buy' implies buying credit risk, where the fixed coupon is received
+   * and the protection is paid  in the event of default.
+   * A value of 'Sell' implies selling credit risk, where the fixed coupon is paid
+   * and the protection is received in the event of default. 
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final BuySell buySell;
+  /**
+   * The CDS index identifier.
+   * <p>
+   * This identifier is used for referring this CDS index product.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final StandardId cdsIndexId;
+  /**
+   * The legal entity identifiers.
+   * <p>
+   * This identifiers are used for the reference legal entities of the CDS index.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableList<StandardId> legalEntityIds;
+  /**
+   * The periodic payments based on the fixed rate.
+   * <p>
+   * Each payment period represents part of the life-time of the leg.
+   * In most cases, the periods do not overlap. However, since each payment period
+   * is essentially independent the data model allows overlapping periods.
+   */
+  @PropertyDefinition(validate = "notEmpty")
+  private final ImmutableList<CreditCouponPaymentPeriod> periodicPayments;
+  /**
+   * The protection end date.
+   * <p>
+   * This may be different from the accrual end date of the last payment period in {@code periodicPayments}.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate protectionEndDate;
+  /**
+   * The day count convention.
+   * <p>
+   * This is used to convert dates to a numerical value.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DayCount dayCount;
+  /**
+   * The payment on default.
+   * <p>
+   * Whether the accrued premium is paid in the event of a default.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final PaymentOnDefault paymentOnDefault;
+  /**
+   * The protection start of the day.
+   * <p>
+   * When the protection starts on the start date.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ProtectionStartOfDay protectionStart;
```

Comment:
This will be used in the method calculateEffectiveStartDate(LocalDate), which is called from the pricer.